grafana-agent: 0.30.2 -> 0.32.1, remove obsolete test fix; nixos/grafana-agent: use `lib.getExe`

emilylange opened the pull request:
###### Description of changes

breaking: 0.31.0 renamed the binaries from `agent` and `agentctl` to `grafana-agent` and `grafana-agentctl`

https://github.com/grafana/agent/releases/tag/v0.32.1
https://github.com/grafana/agent/releases/tag/v0.32.0
https://github.com/grafana/agent/releases/tag/v0.31.3
https://github.com/grafana/agent/releases/tag/v0.31.2
https://github.com/grafana/agent/releases/tag/v0.31.1
https://github.com/grafana/agent/releases/tag/v0.31.0
diff: https://github.com/grafana/agent/compare/v0.30.2...v0.32.1

This version bump will break `services.grafana-agent.package` on 22.11 when `cfg.package` is coming from unstable!

Also removed the obsolete test fix in the `preBuild` hook, as `node_exporter_test.go` does no longer need access to `/sys`.


`services.grafana-agent` has been switched to use `lib.getExe` instead of the hardcoded `/bin/agent` paths (because those paths had to be fixed anyway).

A subset of this pull request could be backported to 22.11 to use `lib.getExe` as well, together with `pkgs.grafana-agent.meta.mainProgram = "agent"` for v0.30.2, so a user setting `cfg.package` to the version from unstable won't break their setup (while keeping the default `cfg.package` compatible as well).
Let me know if I should do that.
I just expect such errors are expected when mixing 22.11 with unstable :woman_shrugging: 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

IndeedNotJames commented:
Not sure why darwin produces a different `vendorHash`.
Will mark this PR as draft until I find time to investigate (and hopefully fix it) :/

zowoq commented on pkgs/servers/monitoring/grafana-agent/default.nix:
```diff
@@ -2,16 +2,16 @@
   };
 
-  vendorHash = "sha256-Cl3oygH1RPF+ZdJvkDmr7eyU5daxaZwNE8pQOHK/qP4=";
+  vendorHash = "sha256-TfXVnTM2rG7cvUV5fqD+tglg16scFvwpt0/D92Xvjtg=";
 
```
```suggestion
  vendorHash = "sha256-5beHG1mZkNcDhccleqTlMA+uiV5d6SSh2QWiL4g3O28=";
  proxyVendor = true; # darwin/linux hash mismatch

```
https://github.com/NixOS/nixpkgs/blob/fc861d3e06657c4a43c958333f8aaa812b359649/doc/languages-frameworks/go.section.md?plain=1#L22

IndeedNotJames replied:
Thank you! :)

emilylange committed:
grafana-agent: 0.30.2 -> 0.32.1

breaking: 0.31.0 renamed the binaries from `agent` and `agentctl` to `grafana-agent` and `grafana-agentctl`

A follow-up commit patches `services.grafana-agent` to use `lib.getExe` instead of a hardcoded `/bin/agent`.

This version bump will break `services.grafana-agent.package` on 22.11 when `cfg.package` is coming from unstable

https://github.com/grafana/agent/releases/tag/v0.32.1
https://github.com/grafana/agent/releases/tag/v0.32.0
https://github.com/grafana/agent/releases/tag/v0.31.3
https://github.com/grafana/agent/releases/tag/v0.31.2
https://github.com/grafana/agent/releases/tag/v0.31.1
https://github.com/grafana/agent/releases/tag/v0.31.0
diff: https://github.com/grafana/agent/compare/v0.30.2...v0.32.1

emilylange committed:
grafana-agent: add indeednotjames as maintainer

emilylange committed:
nixos/grafana-agent: use `lib.getExe` as binary names changed between updates

and `lib.getExe` allows a safe handling and potential backport of this.
But for that to work it would require 22.11 to set `pkgs.grafana-agent.meta.mainProgram = "agent"`

Relevant upstream release: https://github.com/grafana/agent/releases/tag/v0.31.0

emilylange committed:
grafana-agent: remove obsolete test fix in `preBuild` hook

as `node_exporter_test.go` does no longer need access to `/sys`
